miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/storage/MiniAppManifestCache.kt

```diff
@@ -0,0 +1,64 @@
+package com.rakuten.tech.mobile.miniapp.storage
+
+import android.annotation.SuppressLint
+import android.content.Context
+import android.content.SharedPreferences
+import androidx.annotation.VisibleForTesting
+import com.google.gson.Gson
+import com.google.gson.reflect.TypeToken
+import com.rakuten.tech.mobile.miniapp.MiniAppManifest
+import java.lang.Exception
+
+/**
+ * A caching class to read and store the [MiniAppManifest] per MiniApp
+ * using [SharedPreferences].
+ */
+// TODO: unit testcases
+internal class MiniAppManifestCache(context: Context) {
+    private val prefs: SharedPreferences = context.getSharedPreferences(
+        "com.rakuten.tech.mobile.miniapp.manifest.cache", Context.MODE_PRIVATE
+    )
+
+    @VisibleForTesting(otherwise = VisibleForTesting.NONE)
+    fun doesDataExist(miniAppId: String) = prefs.contains(miniAppId)
+
+    /**
+     * Reads the grant results from SharedPreferences.
+     * @param [miniAppId] the key provided to find the stored manifest per MiniApp.
+     * @return [MiniAppManifest] an object to contain the manifest per MiniApp.
+     * if data has been stored in cache, otherwise empty value.
+     */
+    @SuppressLint("RestrictedApi")
+    fun readMiniAppManifest(miniAppId: String): MiniAppManifest {
+        val empty = MiniAppManifest(emptyList(), emptyList(), emptyMap())
+        return if (doesDataExist(miniAppId)) {
+            try {
+                val cachedManifest: MiniAppManifest = Gson().fromJson(
+                    prefs.getString(miniAppId, ""),
+                    object : TypeToken<MiniAppManifest>() {}.type
+                )
```

Comment:
Maybe the retrieval from `prefs` should be moved outside of this `if` statement? It would make the flow a little more clear. For example (including the change to a nullable return type):
```suggestion
        val manifestJsonStr = prefs.getString(miniAppId, null) 
            ?: return empty
        return try {
            val cachedManifest: MiniAppManifest = Gson().fromJson(
                manifestJsonStr,
                object : TypeToken<MiniAppManifest>() {}.type
            )
```

Reply:
I have updated this part with moving it to `MiniAppManifestCache`.